src/Adapter/Category/CategoryDataProvider.php

```diff
@@ -69,4 +70,32 @@ public function getAllCategoriesName($root_category = null, $id_lang = false, $a
         array_shift($categories);
         return $categories;
     }
+
+    /**
+     * return a simple array id/fullpath/name of categories for a specified product
+     * @param Product $product
+     *
+     * @return array Categories
+     */
+    public function getCategoriesByProduct(ObjectModel $product)
```

Comment:
using `Product` instead make PrestaShop cry : "Product expected and ProductCore provided" :/